### PR TITLE
[4.12] Change repo names to match the RHCOS repo names

### PR DESCRIPTION
**Pull request by ravanelli**
- We are changing the repo names in order to improve our internal redhat-coreos repo, to use a single branch for the RHCOS versions.

Signed-off-by: Renata Ravanelli <rravanel@redhat.com>

### CODE REVIEW

**Comment by ravanelli** [comment]
/do-not-merge/hold   

Wait to merge. We will need to change the CI around to not break it.

**Comment by travier** [comment]
Pending openshit/release and internal changes

**Comment by travier** [comment]
We'll need a similar change for 9.0 repos and 4.13 too

**Comment by travier** [comment]
/lgtm cancel
Let's bundle them all in one PR?

**Review by jlebon — APPROVED** [review]

**Comment by travier** [comment]
We either have to fix the repo names in https://github.com/openshift/release/blob/master/core-services/release-controller/_repos/ocp-4.13-rhel86.repo & https://github.com/openshift/release/blob/master/core-services/release-controller/_repos/ocp-4.13-rhel90.repo or sed them in the script here to match what we want.

**Comment by jlebon** [comment]
So now that we've branched for 4.12, I think one way to ratchet this in is:
1. Merge the internal redhat-coreos MR that updates the repo definitions; this won't break 4.12 and earlier since they're using a separate branch of that repo. It theoretically breaks 4.13 but that doesn't exist yet.
2. Change the yum repo names referenced in `master` of this repo to the new 4.13 names.
3. Verify 4.13 in the new pipeline is happy.
4. When we're ready to move 4.12 to the new pipeline, change the yum repo names referenced in `release-4.12` of this repo to the new 4.12 names and verify 4.12 in the new pipeline is happy.
5. Repeat 4 for 4.11, then 4.10, etc...

Worst-case scenario, if we have to flip back to the old pipeline for e.g. 4.12, we can update the yum repo names in the `4.12` branch of the internal redhat-coreos repo.

**Comment by jlebon** [comment]
> We either have to fix the repo names in [openshift/release@`master`/core-services/release-controller/_repos/ocp-4.13-rhel86.repo](https://github.com/openshift/release/blob/master/core-services/release-controller/_repos/ocp-4.13-rhel86.repo?rgh-link-date=2022-11-18T19%3A44%3A31Z) & [openshift/release@`master`/core-services/release-controller/_repos/ocp-4.13-rhel90.repo](https://github.com/openshift/release/blob/master/core-services/release-controller/_repos/ocp-4.13-rhel90.repo?rgh-link-date=2022-11-18T19%3A44%3A31Z) or sed them in the script here to match what we want.

I'm not too sure what else uses those repos, but if it's safe to rename them there, then that'd be great. The files already seem versioned by RHEL and OCP versions. Eventually, we can rename it back to e.g. `rhel-8.6-server-ose` when we start using [`repovars`](https://github.com/coreos/rpm-ostree/pull/4152) for the OCP release bit.

**Comment by travier** [comment]
The RHEL versioned `openshift/release` repos should only be used by us so we can change them when we're ready to move forward with this one. The PRs there don't need passing CI here so we're good to merge there first then here over green CI.

File: ci/prow-entrypoint.sh [hunk 1]
```diff
@@ -86,14 +86,14 @@ prepare_repos() {
 
         # Temporary workaround until we have all packages for RHCOS 9
         curl --fail -L "http://base-${ocpver_mut}-rhel86.ocp.svc.cluster.local" -o "src/config/tmp.repo"
-        awk '/rhel-8-server-ose/,/^$/' "src/config/tmp.repo" > "src/config/ocp86.repo"
+        awk '/rhel-8.6-server-ose-4.12/,/^$/' "src/config/tmp.repo" > "src/config/ocp86.repo"
```
[the remark this answers is not in the document]
Comment (travier) [reply ci/prow-entrypoint.sh:89]: Does https://github.com/openshift/os/pull/1066#issuecomment-1321758660 answer that?

[the remark this answers is not in the document]
Comment (jlebon) [reply ci/prow-entrypoint.sh:89]: Yup, it does.

**Review by jlebon — APPROVED** [review]
/lgtm

**Comment by travier** [comment]
We'll need an override from `@mrunalp` or `@aravindhp` once we're ready for this one (and the other repo related ones) if we don't want to make a bug chain.

**Comment by jlebon** [comment]
We're going full steam ahead with the new pipeline, so we're ready for this (and the other ones similar to it). @mrunalp Would you be able to stamp this? The TL;DR is: we're tweaking repo names to work better with some pipeline work we're doing. No actual content change should happen here.

**Pull request merged**